application/common/models/User.php

```diff
@@ -350,4 +376,56 @@ public function getVerifiedMethods()
     {
         return Method::findAll(['user_id' => $this->id, 'verified' => 1]);
     }
+
+    public function getPasswordMeta()
+    {
+        $thisUser = self::findOne(['id' => $this->id]);
+        return [
+            'last_changed' => Utils::getIso8601($thisUser->pw_last_changed),
+            'expires' => Utils::getIso8601($thisUser->pw_expires),
+        ];
+    }
+
+    /**
+     * @param string $newPassword
+     * @throws ServerErrorHttpException
+     * @throws \yii\web\BadRequestHttpException
+     */
+    public function setPassword($newPassword)
+    {
+        $password = Password::create($newPassword);
+        $password->save($this->employee_id);
+
+        $this->pw_last_changed = Utils::getDatetime();
+        $this->pw_expires = Utils::getDatetime(time() + \Yii::$app->params['passwordLifetime']);
+        
+        if ( ! $this->save()) {
+            throw new ServerErrorHttpException('Unable to save user profile after password change', 1466104537);
+        }
+    }
+
+    /**
+     * @param string $clientId
+     * @return string
+     * @throws ServerErrorHttpException
+     */
+    public function createAccessToken($clientId)
+    {
+        /*
+             * Create access_token and update user
+             */
+        $accessToken = Utils::generateRandomString(32);
+        /*
+         * Store combination of clientId and accessToken for bearer auth
+         */
+        $this->access_token = $clientId . $accessToken;
```

Comment:
These are pretty minor issues, but it may be worth cleaning this function up a little. For example: 
- Does an access token include the client id? Lines 417/419 and 421 seem to give different answers to that question.
- The first comment should probably be moved out to be the doc. comment's description of the function as a whole. That would also resolve the problem of there being no indication from the "outside" (aka. the function's name and doc. comment) that the function is both creating _and saving_ the access token. That may not matter... but more clarity is almost always helpful.